### PR TITLE
Refactor `sync` command

### PR DESCRIPTION
**Pull request by const-cloudinary**
### Brief Summary of Changes

This is our famous `sync` functionality. 

Please approve that the code is readable.

The code is easier to see in a file:
https://github.com/cloudinary/cloudinary-cli/blob/569d381562cf92974df9574947659b87f20c2efa/cloudinary_cli/modules/sync.py

Most of the logic is in the file, some small helper methods are extracted to a separate utils.

If something's not clear enough, or requires more comments/better names, please let me know.

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [x] Refactoring
- [ ] New feature
- [ ] Bug fix
- [ ] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

File: cloudinary_cli/modules/sync.py [hunk 1]
```diff
@@ -160,6 +146,36 @@ def push(self):
         if upload_errors:
             raise Exception("Sync did not finish successfully")
 
+    def pull(self):
```
Comment (strausr) [inline cloudinary_cli/modules/sync.py:149]: we're returning booleans do you think we should change the name of this method to represent this?

File: cloudinary_cli/modules/sync.py [hunk 2]
```diff
@@ -160,6 +146,36 @@ def push(self):
...
+        """
+        Pulls changes from the Cloudinary folder to the local folder.
+        """
+        download_results = {}
+        download_errors = {}
+        if not self._handle_unique_local_files():
+            return False
```
Comment (strausr) [inline cloudinary_cli/modules/sync.py:156]: log error for all returns?

File: cloudinary_cli/modules/sync.py [hunk 3]
```diff
@@ -160,6 +146,36 @@ def push(self):
...
+
+        files_to_pull = self.unique_remote_file_names | self.out_of_sync_remote_file_names
```
Comment (strausr) [inline cloudinary_cli/modules/sync.py:158]: log from which source we're pulling

File: cloudinary_cli/modules/sync.py [hunk 4]
```diff
@@ -160,6 +146,36 @@ def push(self):
...
+
+        if not files_to_pull:
+            return True
+
+        logger.info(f"Downloading {len(files_to_pull)} files from Cloudinary")
+        downloads = []
+        for file in files_to_pull:
+            remote_file = self.remote_files[file]
+            local_path = path.abspath(path.join(self.local_dir, file))
```
Comment (strausr) [inline cloudinary_cli/modules/sync.py:167]: why are we naming this local_path when we're using the abspath method?

File: cloudinary_cli/modules/sync.py [hunk 5]
```diff
@@ -190,6 +206,12 @@ def _save_sync_meta_file(self, upload_results):
                 logger.warning(f"Failed updating '{self.sync_meta_file}' file: {e}")
```
Comment (strausr) [inline cloudinary_cli/modules/sync.py:206]: is this indentation on purpose

**Review by strausr — APPROVED** [review]

File: cloudinary_cli/modules/sync.py (hunk 1 above)
Comment (const-cloudinary) [reply cloudinary_cli/modules/sync.py:149]: @strausr , in python it's not common to prefix functions with `is_` or something like that.

File: cloudinary_cli/modules/sync.py (hunk 2 above)
Comment (const-cloudinary) [reply cloudinary_cli/modules/sync.py:156]: @strausr everything's being logged inside the function itself.

File: cloudinary_cli/modules/sync.py (hunk 3 above)
Comment (const-cloudinary) [reply cloudinary_cli/modules/sync.py:158]: @strausr , it is logged at the very beginning of the execution of the `sync` command

File: cloudinary_cli/modules/sync.py (hunk 4 above)
Comment (const-cloudinary) [reply cloudinary_cli/modules/sync.py:167]: @strausr, here `local` means that it is on the current file system and not `remote` on cloudinary.

File: cloudinary_cli/modules/sync.py (hunk 5 above)
Comment (const-cloudinary) [reply cloudinary_cli/modules/sync.py:206]: @strausr yes!

**Pull request merged**